converter/generate_image.py

```diff
@@ -1,13 +1,65 @@
-from flask import current_app, Blueprint, send_file, flash, redirect
+import os
+from flask import current_app, Blueprint, flash, send_file, redirect
 from music21 import *
+from zipfile import ZipFile
+
 bp = Blueprint("get_image", __name__)
 
 #download result image
-@bp.route('/get-image/')
+@bp.route('/get-image/', methods=['GET', 'POST'])
 def generate_image():
     try:
-        return send_file(converter.parse('result.abc').write('musicxml.png'), 
-            attachment_filename="result.png" ,as_attachment=True, cache_timeout=0)
+        #create zip file for download
+        zipObj = ZipFile(current_app.config['OUTPUT_ZIP'], 'w')
+        #get temp result images path
+        result_image_path = converter.parse(current_app.config['PROCESS_FILE']).write('musicxml.png')
+        
+        #detect for multiple outputs
+        #1 < image num < 10
+        if '-1.png' in result_image_path:
+            image_temp_prefix = result_image_path[:-6]
+            for i in range(1,10):
+                try:
+                    zipObj.write(image_temp_prefix+'-'+str(i)+'.png')
+                except:
+                    break
+        #10 < image num < 100
+        elif '-01.png' in result_image_path:
+            image_temp_prefix = result_image_path[:-7]
```

Comment:
This whole section below could be much simplified as all the duplicated code is just meant to pad zeros to the generated file name (besides some formatting beauty what is the point of doing that?). 

All you need to do is have one loop like the following:

```
max_image_number = 999

for i in range(max_image_number):
    image_suffix = calc_suffix(i);
    image_name = image_temp_prefix + image_suffix + ".png"
    try:
        zipObj.write(image_name);
    except:
        break
```

where `calc_suffix(i)` gives you the proper suffix string by checking the range of i. I'd suspect one can put out a one-liner for this calc_suffix calculation even. 

Please rework this part and get it clean.

Reply:
implemented